Add libkrun Mac task

cevich opened the pull request:
***Depends on:*** https://github.com/containers/automation/pull/208 + deployment into worker pool.

#### Does this PR introduce a user-facing change?

```release-note
None
```

cevich commented:
[Initial test-run](https://cirrus-ci.com/build/4881385851191296)

cevich commented:
Testing note: Confirmed (expected) libkrun tests all fail when run on Mac's w/o libkrun.

cevich commented:
Opened https://github.com/containers/podman/issues/23296

cevich commented:
@edsantiago and/or @baude if you want to give the first two commits an initial look over, this PR is code-complete once https://github.com/containers/automation/pull/208 is merged and deployed to the pool management VM (after I'm back from PTO).

cevich commented:
FYI- I'll be off on PTO next week, so won't be enabling libkrun testing in Podman until I return.  There are [changes needed on the Mac's](https://github.com/containers/automation/pull/208) used for CI and I don't want to risk breaking something while I'm away.

edsantiago commented on contrib/cirrus/mac_runner.sh:
```diff
@@ -29,4 +29,22 @@ function logformatter() {
+        # Otherwise if the var. goes out-of-scope, defaults change, or definition
+        # fails: Suddenly assumed behavior != actual behaviorr, esp. if/when only
+        # quickly glancing at a green status check-mark.
+        export CONTAINERS_MACHINE_PROVIDER=""
```
It makes me very uncomfortable to have an envariable set to empty. More so when the previous behavior was to have it set to `applehv`. Is it possible to make it `applehv` here? I see that it's consumed in `pkg/machine/provider/*` but haven't chased donw how it handles "" vs "actualvalue"

cevich replied:
Yes, makes me uncomfortable as well.  Unfortunately I literally copy/pasted most of that `FIXME` from our windows setup, which does the exact same thing :disappointed: 

Thanks for checking the code.  I'll try setting it to `applehv`, though undocumented, maybe it will 'just work'?

cevich replied:
Trolled through the code a bit.  It appears the default comes from containers-common.  Setting "applehv" or "libkrun" overrides it, then the value is checked against a list of static strings to catch bad names.  So this should be good (and I'll give it a try).

cevich replied:
[This works (:sweat:) thankfully](https://api.cirrus-ci.com/v1/artifact/task/6149203389317120/html/mac-applehv-podman-darwin-rootless-host-sqlite.log.html).  Perhaps it will inspire a similar fix for Windows testing :grin:

edsantiago reviewed (commented):
Tentative LGTM with two concerns

cevich commented:
[The libkrun mac tests are also passing](https://cirrus-ci.com/task/5586253435895808).  I will merge https://github.com/containers/automation/pull/208 shortly, then after a day or so this PR can have final review and testing performed (i.e. after the pool has updated).

cevich commented:
This PR can move forward on/after 2024-08-01T18:50:12+00:00, by then all the Macs in the pool will be running the new setup.

cevich commented:
Force-push:
* Removed `DO NOT MERGE` commit to clear up diff (this still isn't ready to merge).
* Renamed mac `TEST_FLAVOR` values to be consistent w/ other podman-machine testing tasks.
  (so they all sort together in the status page).

cevich commented:
This is ready for final review and (assuming CI passes) merging.

cevich commented:
Looks like the MacOS PM tests are both green, [applehv](https://api.cirrus-ci.com/v1/artifact/task/5284978806751232/html/machine-applehv-podman-darwin-rootless-host-sqlite.log.html) and [libkrun](https://api.cirrus-ci.com/v1/artifact/task/6410878713593856/html/machine-libkrun-podman-darwin-rootless-host-sqlite.log.html)

rhatdan commented:
LGTM
@mheon @baude @Luap99 PTAL

mheon commented:
/lgtm

rhatdan commented:
/approve
